qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,335 @@
+                rst_oper = Reset()
+                if is_rst_control:
+                    qc._append(
+                        CircuitInstruction(
+                            operation=rst_oper,
+                            qubits=[qubits[control_qubit]],
+                        )
+                    )
+                if is_rst_target:
+                    qc._append(
+                        CircuitInstruction(
+                            operation=rst_oper,
+                            qubits=[qubits[target_qubit]],
+                        )
+                    )
+
+            if current_instr.is_parameterized():
+                # Check if instruction is mutable, so that it could accept parameters.
+                if not current_instr.mutable:
+                    current_instr = current_instr.to_mutable()
+                params = parameters[:num_gate_params]
+                parameters = parameters[num_gate_params:]
+                current_instr.params = params
+
+            if is_cond and conditional:
+                qc.measure(qc.qubits, cr)
+                # The condition values are required to be bigints, not Numpy's fixed-width type.
+                current_instr = current_instr.c_if(cr, int(condition_values[clbit_idx]))
+                clbit_idx += 1
+
+            if control_qubit in qubit_idx_not_used and target_qubit in qubit_idx_not_used:
+                qc._append(
+                    CircuitInstruction(
+                        operation=current_instr,
+                        qubits=[qubits[control_qubit], qubits[target_qubit]],
+                    )
+                )
+
+                qubit_idx_used.update(set(edge))
+                qubit_idx_not_used = qubit_idx_not_used - qubit_idx_used
+                edges_used[(control_qubit, target_qubit)] += 1
+
+        if insert_1q_oper:
+            if not len(qubit_idx_not_used) == 0:
+
+                # Some extra 1Q Gate in to fill qubits which are still idle for this
+                # particular while iteration.
+                extra_1q_gates = rng.choice(gates_1q, size=len(qubit_idx_not_used), replace=True)
+
+                cumsum_params = np.cumsum(extra_1q_gates["num_params"], dtype=np.int64)
+                parameters_1q = rng.uniform(0, 2 * np.pi, size=cumsum_params[-1])
+
+                for current_instr, num_gate_params, qubit_idx in zip(
+                    extra_1q_gates["class"], extra_1q_gates["num_params"], qubit_idx_not_used
+                ):
+
+                    if current_instr.is_parameterized():
+                        # Check if instruction is mutable, so that it could accept parameters.
+                        if not current_instr.mutable:
+                            current_instr = current_instr.to_mutable()
+                        param = parameters_1q[:num_gate_params]
+                        parameters_1q = parameters_1q[num_gate_params:]
+                        if isinstance(current_instr, Delay):
```

Comment:
Not removing `Delay` from 1Q operation, as it could also be applied conditionally?